sensor/kubernetes/listener/resources/virtualmachine/store/virtual_machine_store.go

```diff
@@ -187,8 +187,11 @@ func (s *VirtualMachineStore) replaceVSOCKInfoNoLock(vm *virtualmachine.Info) *u
 		vm.VSOCKCID = prev.VSOCKCID
 	}
 	// Upsert VSOCKCID info
+	// CRITICAL: addOrUpdateVSOCKInfoNoLock always returns a heap-allocated copy so the store owns
+	// its own pointer. Reusing vm.VSOCKCID would let the caller mutate the same pointer later.
+	// Added regression test: Test_replaceVSOCKInfoNoLockCopiesIncomingPointer.
 	if vm.VSOCKCID != nil {
-		_ = s.addOrUpdateVSOCKInfoNoLock(vm.ID, vm.VSOCKCID)
+		return s.addOrUpdateVSOCKInfoNoLock(vm.ID, vm.VSOCKCID)
 	}
 	return vm.VSOCKCID
```

Comment:
How about explicitly return nil when `vm.VSOCKCID == nil`?
```suggestion
	return nil
```

Reply:
You are right, this will always be nil in this place. Returning `vm.VSOCKCID` vs. returning `nil` is a style question only. I would slightly lean towards returning the pointer following the "return the pointer we’re going to store" idea.